### ISSUE TITLE
Field level validation - get name of the currently validated field

### ISSUE THREAD

**Issue by Misiur** [erikras/redux-form#3350]
This is a feature request
From quick glance at redux-form code, currently I see that the name of currently validated field name is discarded. I made the following change:
```diff
diff --git a/src/generateValidator.js b/src/generateValidator.js
index 8a7e503..35bd705 100644
--- a/src/generateValidator.js
+++ b/src/generateValidator.js
@@ -10,11 +10,12 @@ const getError = (
   value: any,
   values: Object,
   props: Object,
-  validators: any
+  validators: any,
+  name: string
 ) => {
   const array = toArray(validators)
   for (let i = 0; i < array.length; i++) {
-    const error = array[i](value, values, props)
+    const error = array[i](value, values, props, name)
     if (error) {
       return error
     }
@@ -28,7 +29,7 @@ const generateValidator = (
   let errors = {}
   Object.keys(validators).forEach(name => {
     const value = getIn(values, name)
-    const error = getError(value, values, props, validators[name])
+    const error = getError(value, values, props, validators[name], name)
     if (error) {
       errors = plain.setIn(errors, name, error)
     }
```

All mocha tests succeed, other tests are stuck on my computer. Shall I make a PR, or you are against changing API?

**Comment by gustavohenke** [comment]
Done by #3364

**Comment by erikras** [comment]
Published fix in [`v7.1.0`](https://github.com/erikras/redux-form/releases/tag/v7.1.0).
